pandas/core/indexing.py

```diff
@@ -1880,15 +1928,26 @@ def _has_valid_type(self, key, axis):
                                               "is not available")
                 raise ValueError("iLocation based boolean indexing cannot use "
                                  "an indexable as a mask")
-            return True
+            return
 
         if isinstance(key, slice):
-            return True
+            return
         elif is_integer(key):
-            return self._is_valid_integer(key, axis)
+            self._validate_integer(key, axis)
+        elif isinstance(key, tuple):
+            # a tuple should already have been caught by this point
+            # so don't treat a tuple as a valid indexer
+            raise IndexingError('Too many indexers')
```

Comment:
this is hit in tests?

Reply:
Sure: https://github.com/pandas-dev/pandas/blob/6cacdde5630c593999059833b516e1fec60aaf72/pandas/tests/frame/test_indexing.py#L1389

(this is old code I just moved around)